src/DirectGateway.php

```diff
@@ -43,6 +44,16 @@ public function setSimulatorMode($value)
         return $this->setParameter('simulatorMode', $value);
     }
 
+    public function getReferrerId()
+    {
+        return $this->getParameter('referrerId');
+    }
+
+    public function setReferrerId($value)
+    {
+        return $this->setParameter('referrerId', $value);
+    }
+
```

Comment:
Do we need the functions on both request and gateway? Looks like the tests are only covering the request usage